### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by obibaadoma**
# Project Requirements

## `app.rb` Console Application

1. Create an `app.rb` file as the console app entry point.

## Methods in `app.rb`

2. Implement methods in `app.rb` to perform the following actions:
   - List all books.
   - List all people.
   - Create persons (teachers/students).
   - Create a book.
   - Create a rental.
   - List rentals for a given person ID.

## `main.rb` Entry Point

3. Define the `main` method in `main.rb` for user interaction.
   - Present a list of options to the user.
   - Allow users to choose an option.
   - If necessary, collect input parameters.
   - Provide a way to gracefully quit the app.

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Derek, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your ReadMe is professionally documented; The _Key Features_ section of your ReadMe should include features related to your project. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  puts 'Welcome to library Application!'
+  loop do
+    print_question
+    selected_option = app.select_opt
+    break if selected_option == 7
+  end
+  puts 'Thanks for using this App!!'
+end
+
+Main
```
Comment (DammyShittu) [inline main.rb]: - [ ] Kindly run your app and ensure it works before submitting it for review. None of these options works at all. From options 1 to 6. Only exiting the app, option 7, works.

  I cannot create people, books or rentals. I cannot list any of them either. None works at all. See example of error message.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @obibaadoma,

## Highlights
✔️ App exits

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Alusp — CHANGES_REQUESTED** [review]
Hi @obibaadoma , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
close

 ## Highlights
✔️ Good PR Summary. ✔️
 ✔️ Completed most of the TASKS ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alusp in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+      puts "Date: #{rental.date} Book: #{rental.book.title} by #{rental.book.author}" if rental.person.id == id_person
+    end
+  end
+
+  def main
+    loop do
+      puts 'Options:'
+      puts '1. List books'
+      puts '2. List people'
+      puts '3. Create person'
```
Comment (Alusp) [inline app.rb]: - [ ] Good job implementing the change requested by the previous reviewer, however, I ran into an error after running your project, please resolve it. :slightly_smiling_face: 

Re_Review

**Review by gemmen29 — APPROVED** [review]
Hi @obibaadoma, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**